core/src/main/java/org/elasticsearch/discovery/zen/fd/NodesFaultDetection.java

```diff
@@ -275,8 +280,8 @@ public void messageReceived(PingRequest request, TransportChannel channel) throw
 
     public static class PingRequest extends TransportRequest {
 
-        // the (assumed) node id we are pinging
-        private String nodeId;
+        // the (assumed) node we are pinging
+        private DiscoveryNode node;
```

Comment:
maybe call this `targetNode` so it is clearer that it's the node we are pinging?